tests/trace/test_weave_client.py

```diff
@@ -1886,3 +1887,45 @@ def my_op(a: int) -> int:
 
         # Local attributes override global ones
         assert call.attributes["env"] == "override"
+
+
+def test_flush_progress_bar(client):
```

Comment:
this is not testing the pbar.

If you really want to test the pbar, you can capture redirect IO into a buf and confirm it looks the way you want

Reply:
hmmm yeah, the issue is actually more generating a test that actually requires flushing. not sure there is a trivial way of doing it... scratching my head over here

Reply:
There are a few ways, but maybe the easiest is to use a mock server that is slow to respond / only accepts 1 connection.  Then you can queue up some jobs and see them go through 1-by-1